xcm/pallet-xcm/src/lib.rs

```diff
@@ -699,7 +699,7 @@ pub mod pallet {
 							xcm: Xcm(vec![]),
 						},
 					]);
-					T::Weigher::weight(&mut message).map_or(Weight::MAX, |w| Weight::from_ref_time(100_000_000.saturating_add(w)))
```

Comment:
@gavofyork The XCM pallet still contains calls to `from_ref_time` in places such as this. The intent here is to account for the weight used for executing the code in the extrinsic _without_ the weight from executing the XCM instructions. I suspect that this is not trivial to benchmark, hence the pseudo value of 100m being used here. As such, I don't believe that it makes sense to use `from_parts` here, but am unsure what the proper fix is, or whether I should just leave it alone.

Reply:
This function needs weighing properly as the 100_000_000 is just a conservative guess.